plugins/conversion.js

```diff
@@ -0,0 +1,99 @@
+const _ = require('lodash');
+
+module.exports = (function() {
+
+	let bot;
+
+	function doConversion (message, channel) {
+
+		let [command, ...args] = message.parts;
+		let measurement, unit;
+
+		if(!args.length){
+			channel.send('Please provide something to convert.');
+			return true
+		}
+
+		//The user put a space between the measurement and the units
+		if(args.length > 1){
+			measurement = args[0];
+			unit = args[1].toLowerCase();
+		} else {
+
+			//Extract the measurement and units from a single string
+			let measurements = args[0].match(/^[\d.-]+/g);
+			let units = args[0].match(/[^\d.-]+/g);
+
+			if( !(measurements && measurements.length && units && units.length) ){
+				channel.send( 'Unable to extract measurement and/or units from the given parameters');
+			} else {
+				measurement = measurements[0];
+				unit = units[0].toLowerCase();
+			}
+
+		}
+
+		if( ( unit !== 'f' && unit !== 'c' ) && measurement < 0 ){
+			channel.send( 'Can\'t go below 0 on that one mate...' );
+			return true;
+		}
+
+		switch (unit) {
+			case 'f':
+				channel.send( measurement + ' degrees fahrenheit is approximately ' + convertFahrenheitToCelcius(measurement) + ' degrees celcius.' );
```

Comment:
consider using the stringTemplates for this, might be a bit more readable. you could also use the degree symbol °

```js
channel.send(`${measurement}°F is approximately ${convertFahrenheitToCelcius(measurement)}°C.`);
```